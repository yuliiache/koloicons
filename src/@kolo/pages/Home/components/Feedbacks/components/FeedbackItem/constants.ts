export const FEEDBACK_TEXT_TEST_ID = 'feedbackText';
