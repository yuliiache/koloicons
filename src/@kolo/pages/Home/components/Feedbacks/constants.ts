export const FEEDBACKS_KEY = 'feedbacks';
