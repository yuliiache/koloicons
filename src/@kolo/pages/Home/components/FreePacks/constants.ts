export const FREE_PACKS_KEY = 'freePacks';
export const FREE_PACKS_TEST_ID = 'packItemComponent';
