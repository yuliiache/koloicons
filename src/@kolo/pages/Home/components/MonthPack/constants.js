export const MONTH_PACK_KEY = 'monthPack';
