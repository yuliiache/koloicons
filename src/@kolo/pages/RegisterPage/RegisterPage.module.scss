@import "@kolo/design";

.main {
  display: flex;
  flex-direction: column;
  min-height: calc(100vh - 144px);

  @media (max-width: $mobileWidth) {
    min-height: calc(100vh - 64px);
  }
}

.register {
  &Section {
    padding: 32px 0;
    background-color: $DarkBG;

    @media (max-width: $mobileWidth) {
      padding: 20px 0 32px;

      background-color: $LightBG;
    }
  }

  &Container {
    margin: 0 auto;
    max-width: 448px;
    padding: 24px;
    background-color: $LightBG;
    border-radius: 4px;
    box-shadow: $FormBoxShadow;

    @media (max-width: $mobileWidth) {
      max-width: 480px;
      width: 100%;
      padding: 0 16px;
      box-shadow: none;
    }
  }

  &Title {
    margin-bottom: 26px;
    text-align: center;

    font-size: 0.8rem;
    color: $secFontColor;

    @media (max-width: $mobileWidth) {
      margin-bottom: 21px;
    }
  }

  &Text {
    margin-bottom: 10px;

    text-align: center;
    @include main-text(0.6rem, 1.33, null);
    color: $secFontColor;
  }
}
