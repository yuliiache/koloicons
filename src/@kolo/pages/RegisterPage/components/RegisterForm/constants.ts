export const REGISTER_FORM_TEST_ID = 'registerFormComponent';

export const enum CheckboxDataTestId {
  SUBSCRIBE_AGREEMENT_CHECKBOX = 'subscribeAgreementCheckbox',
  TERMS_AGREEMENT_CHECKBOX = 'termsAgreementCheckbox',
}
