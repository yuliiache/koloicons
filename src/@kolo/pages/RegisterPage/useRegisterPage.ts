import {useAppDispatch, useAppSelector} from 'app/hooks';
import {AppRoute} from 'constants/AppRoute';
import {useEffect} from 'react';
import {useNavigate} from 'react-router-dom';

import {isAuthenticatedSelector} from '../LoginPage/selectors';
import {registerUserStartAction} from '../LoginPage/slices/loginUserSlice';
import {RegisterFormValues} from './components/RegisterForm/types/RegisterFormValues';

type HandleSubmit = (value: RegisterFormValues) => void;

interface useRegisterPageReturn {
  handleSubmit: HandleSubmit;
}

const useRegisterPage = (): useRegisterPageReturn => {
  const dispatch = useAppDispatch();
  const navigate = useNavigate();
  const isAuthenticated = useAppSelector(isAuthenticatedSelector);

  useEffect(() => {
    if (isAuthenticated) {
      navigate(AppRoute.ICONS);
    }
  }, [isAuthenticated]);

  const handleSubmit: HandleSubmit = (value) => {
    dispatch(registerUserStartAction(value));
  };

  return {handleSubmit};
};

export default useRegisterPage;
