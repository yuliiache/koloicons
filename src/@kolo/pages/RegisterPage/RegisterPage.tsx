import {ButtonColor, ButtonSize} from '@kolo/uiKit/Button/constants';
import ButtonLink from '@kolo/uiKit/ButtonLink/ButtonLink';
import {AppRoute} from 'constants/AppRoute';
import {FC} from 'react';

import RegisterForm from './components/RegisterForm/RegisterForm';
import {RegisterFormValues} from './components/RegisterForm/types/RegisterFormValues';
import styles from './RegisterPage.module.scss';

interface FormValues {
  firstName: string;
  lastName: string;
  email: string;
  password: string;
}

const trimFormValues = (values: FormValues): FormValues => {
  return {
    firstName: values.firstName.trim(),
    lastName: values.lastName.trim(),
    email: values.email.trim(),
    password: values.password.trim(),
  };
};

const RegisterPage: FC = () => {
  const onFormSubmit = (values: RegisterFormValues) => {
    const trimmedValues = trimFormValues(values);

    console.log('RegisterPage, trimmed user data:', {...values, ...trimmedValues});
  };

  return (
    <main className={styles.main}>
      <section className={styles.registerSection}>
        <div className={styles.registerContainer}>
          <h2 className={styles.registerTitle}>Sign Up</h2>

          <RegisterForm onSubmit={onFormSubmit} />

          <div>
            <p className={styles.registerText}>Already registered?</p>
            <ButtonLink
              text="Sign In"
              size={ButtonSize.FULL_WIDTH}
              color={ButtonColor.BASIC}
              url={AppRoute.LOGIN}
            />
          </div>
        </div>
      </section>
    </main>
  );
};

export default RegisterPage;
