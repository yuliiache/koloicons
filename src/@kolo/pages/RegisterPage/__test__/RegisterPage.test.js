import {BrowserRouter} from 'react-router-dom';
import {cleanup, fireEvent, render, screen} from '@testing-library/react';
import {AppRoute} from 'constants/AppRoute';

import RegisterPage from '../RegisterPage';

describe('RegisterPage component', () => {
  afterEach(cleanup);
  const onSubmitMock = jest.fn();
  const renderPage = () => render(<RegisterPage onSubmit={onSubmitMock} />, {wrapper: BrowserRouter});

  it('should render snapshot successfully', () => {
    const {asFragment} = renderPage();

    expect(asFragment()).toMatchSnapshot();
  });

  it('should navigate to login page when user click the Sign In button', () => {
    renderPage();
    const signInButton = screen.getByText('Sign In');

    fireEvent.click(signInButton);

    expect(signInButton).toBeInTheDocument();
    expect(window.location.pathname).toEqual(AppRoute.LOGIN);
  });
});
