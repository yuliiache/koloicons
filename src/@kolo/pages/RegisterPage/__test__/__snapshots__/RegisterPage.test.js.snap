// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage component should render snapshot successfully 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <section
      class="registerSection"
    >
      <div
        class="registerContainer"
      >
        <h2
          class="registerTitle"
        >
          Sign Up
        </h2>
        <div
          data-testid="registerFormComponent"
        >
          <form
            class="registerForm"
          >
            <div
              class="inputGroup"
            >
              <label
                class="inputGroupLabel"
                for="firstName"
              >
                First name
              </label>
              <input
                class="inputGroupInput"
                id="firstName"
                type="text"
                value=""
              />
            </div>
            <div
              class="inputGroup"
            >
              <label
                class="inputGroupLabel"
                for="lastName"
              >
                Last name
              </label>
              <input
                class="inputGroupInput"
                id="lastName"
                type="text"
                value=""
              />
            </div>
            <div
              class="inputGroup"
            >
              <label
                class="inputGroupLabel"
                for="email"
              >
                Email
              </label>
              <input
                class="inputGroupInput"
                id="email"
                type="text"
                value=""
              />
            </div>
            <div
              class="inputGroup"
            >
              <label
                class="inputGroupLabel"
                for="password"
              >
                Password
              </label>
              <input
                class="inputGroupInput"
                id="password"
                type="password"
                value=""
              />
            </div>
            <div
              class="checkboxWrapper"
            >
              <div
                data-testid="subscribeAgreementCheckbox"
              >
                <label
                  class="checkbox"
                >
                  <input
                    checked=""
                    class="checkbox__input"
                    id="subscribeAgreement"
                    type="checkbox"
                  />
                  <span
                    class="checkbox__indicator"
                  />
                </label>
              </div>
              <p
                class="checkboxText"
              >
                Agree to receive emails from koloicons with discounts or special offers.
              </p>
            </div>
            <div
              class="checkboxWrapper"
            >
              <div
                data-testid="termsAgreementCheckbox"
              >
                <label
                  class="checkbox"
                >
                  <input
                    class="checkbox__input"
                    id="termsAgreement"
                    type="checkbox"
                  />
                  <span
                    class="checkbox__indicator"
                  />
                </label>
              </div>
              <p
                class="checkboxText textDesktop"
              >
                Agree to KOLO Icons Terms of Use and Privacy Policy.
              </p>
              <p
                class="checkboxText textMobile"
              >
                Agree to KOLO Icons 
                <a
                  class="link link--default"
                  data-testid="internalLinkComponent"
                  href="/terms-of-service"
                >
                  <span>
                    Terms of Use
                  </span>
                </a>
                 and 
                <a
                  class="link link--default"
                  data-testid="internalLinkComponent"
                  href="/privacy-policy-docs"
                >
                  <span>
                    Privacy Policy
                  </span>
                </a>
                .
              </p>
            </div>
            <button
              class="button button-full-width button-primary"
              data-testid="buttonComponent"
              type="submit"
            >
              <span
                data-testid="buttonComponentText"
              >
                Sign Up
              </span>
            </button>
          </form>
        </div>
        <div>
          <p
            class="registerText"
          >
            Already registered?
          </p>
          <a
            data-testid="buttonLinkComponent"
            href="/login"
          >
            <button
              class="button button-full-width button-basic"
              data-testid="buttonComponent"
              type="button"
            >
              <span
                data-testid="buttonComponentText"
              >
                Sign In
              </span>
            </button>
          </a>
        </div>
      </div>
    </section>
  </main>
</DocumentFragment>
`;
